examples/getting_started/web_search.ipynb

```diff
@@ -0,0 +1,255 @@
+        "    def _stream(self, query: str): ...\n",
+        "\n",
+        "    def _step(self, query: str):\n",
+        "        stream = self._stream(query)\n",
+        "        tools_and_outputs = []\n",
+        "        for chunk, tool in stream:\n",
+        "            if tool:\n",
+        "                tools_and_outputs.append((tool, tool.call()))\n",
+        "            else:\n",
+        "                print(chunk.content, end=\"\", flush=True)\n",
+        "        if stream.user_message_param:\n",
+        "            self.history.append(stream.user_message_param)\n",
+        "        self.history.append(stream.message_param)\n",
+        "        if tools_and_outputs:\n",
+        "            self.history += stream.tool_message_params(tools_and_outputs)\n",
+        "            self._step(\"\")\n",
+        "\n",
+        "    def run(self):\n",
+        "        while True:\n",
+        "            query = input(\"(User): \")\n",
+        "            if query.lower() in [\"exit\", \"quit\"]:\n",
+        "                break\n",
+        "            print(\"(Assistant): \", end=\"\", flush=True)\n",
+        "            self._step(query)\n",
+        "            print(\"\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "Let's break down the key components of our WebSearchAgent class:\n",
+        "\n",
+        "1. **@openai.call decorator**: This decorator configures the LLM call using OpenAI's model. It specifies the use of the \"gpt-4o-mini\" model, enables streaming, and provides access to the `parse_webpage` and `web_search` tools.\n",
+        "\n",
+        "2. **@prompt_template decorator**: This decorator defines the prompt template sent to the LLM. It allows for dynamic insertion of variables like `{self.max_results}` and `{self.history}`.\n",
+        "\n",
+        "3. **_stream method**: This method generates streaming responses using the OpenAI model. The actual implementation is provided by the Mirascope library, so we don't need to implement the function body.\n",
+        "\n",
+        "4. **_step method**: This method processes a single user query. It handles the streaming response, tool calls, and updates the conversation history.\n",
+        "\n",
+        "5. **run method**: This method provides a simple command-line interface for interacting with the agent. It continuously prompts for user input until the user chooses to exit."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "## 5. Running the Agent <a name=\"running\"></a>\n",
+        "\n",
+        "Now that we have implemented our WebSearchAgent, let's create an instance and run it!"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "agent = WebSearchAgent()\n",
+        "agent.run()"
+      ]
+    },
+    {
```

Comment:
we should actually run the agent and save the output and include the output in the guide so people can see example runs (which is particularly useful when viewing the guide in github)